src/repository.ts

```diff
@@ -958,6 +961,26 @@ export class Repository implements IDisposable {
         return true;
     }
 
+    async shelve(options: ShelveOptions) {
+        return await this.run(Operation.Shelve, () => this.repository.shelve(options));
+    }
+
+    async unshelve(options: UnshelveOptions) {
+        return await this.run(Operation.Shelve, () => this.repository.unshelve(options));
```

Comment:
I'm not very familiar with the Operation thing, but don't we need one for Unshelve?

Reply:
I don't really understand it either. The Git extension uses the same Operation for stash and unstash, so I'm doing the same.